modules/sdk-coin-ada/src/keyPair.ts

```diff
@@ -11,18 +12,51 @@ export class KeyPair extends Ed25519KeyPair {
     super(source);
   }
 
-  getAddress(format?: AddressFormat): string {
-    throw new Error('Method not implemented.');
+  /**
+   *  @param {string} network
+   *  @returns { CardanoWasm.Address }
+   */
+  getAddress(network): string {
+    // enterprise address to start
+    const formattedNetwork = network.toLowerCase().trim();
+    if (formattedNetwork !== 'testnet' || formattedNetwork !== 'mainnet') {
+      throw new Error('Invalid network option');
+    }
+    if (this.keyPair.prv) {
+      const privateKey = CardanoWasm.PrivateKey.from_bech32(this.keyPair.prv); // need to figure out how to create BIP32PrivateKey from here
+      const pubKey = privateKey.to_public();
+      let enterpriseAddress;
+      if (formattedNetwork === 'testnet') {
+        enterpriseAddress = CardanoWasm.EnterpriseAddress.new(
+          CardanoWasm.NetworkInfo.testnet().network_id(),
+          CardanoWasm.StakeCredential.from_keyhash(pubKey.hash())
+        );
+      } else if (formattedNetwork === 'mainnet') {
+        enterpriseAddress = CardanoWasm.EnterpriseAddress.new(
+          CardanoWasm.NetworkInfo.mainnet().network_id(),
+          CardanoWasm.StakeCredential.from_keyhash(pubKey.hash())
+        );
+      }
+      return enterpriseAddress.to_address().to_bech32();
+    } else {
```

Comment:
can we get address from only public key?

Reply:
yes, we need the public key to get the address,

Reply:
but code here is if missing private key, it will throw exception. KeyPair class may only has public key, but should  generate  an address.

Reply:
ohhh I see, yeah that can be changed

Reply:
So if we have just the public key in hex format, that needs to be converted to Cardano bech32 public key for address creation?